components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticator.java

```diff
@@ -0,0 +1,512 @@
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+
+        } else if (request.getParameterMap().containsKey(HYPR.USERNAME)) {
+            // if the login form submission request comes, then go through this flow.
+            initiateHYPRAuthenticationRequest(request, response, context);
+            return AuthenticatorFlowStatus.INCOMPLETE;
+
+        } else if (context.getProperty(HYPR.AUTH_STATUS) != null) {
+            // if intermediate authentication request comes, then go through this flow.
+            String sessionDataKey = request.getParameter(HYPR.SESSION_DATA_KEY);
+            String authStatus = (String) context.getProperty(HYPR.AUTH_STATUS);
+
+            if (authStatus.equals(HYPR.AuthenticationStatus.COMPLETED.getName())) {
+                processAuthenticationResponse(request, response, context);
+                return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.PENDING.getName())) {
+                handleAuthenticationIncompleteState(response, sessionDataKey, HYPR.AuthenticationStatus.PENDING);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.CANCELED.getName())) {
+                handleAuthenticationIncompleteState(response, sessionDataKey, HYPR.AuthenticationStatus.CANCELED);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+
+            } else if (authStatus.equals(HYPR.AuthenticationStatus.FAILED.getName())) {
+                handleAuthenticationIncompleteState(response, sessionDataKey, HYPR.AuthenticationStatus.FAILED);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            }
+        } else {
+            // Redirect the user to the login page.
+            initiateAuthenticationRequest(request, response, context);
+            return AuthenticatorFlowStatus.INCOMPLETE;
+        }
+        return super.process(request, response, context);
+    }
+
+    /**
+     * Retrieve the user session context stored in the HYPR context manager.
+     *
+     * @param response             The response that is received to the authenticator.
+     * @param sessionDataKey       The session data key extracted from the request.
+     * @param authenticationStatus An AuthenticationStatus object which specifies the current status of the
+     *                             authentication
+     * @throws HYPRAuthnFailedException Exception thrown when redirecting the user to login page on which the error
+     *                                  messages are displayed.
+     */
+    private void handleAuthenticationIncompleteState(HttpServletResponse response, String sessionDataKey,
+                                                     HYPR.AuthenticationStatus authenticationStatus)
+            throws HYPRAuthnFailedException {
+
+        redirectHYPRLoginPage(response, sessionDataKey, authenticationStatus);
```

Comment:
Can't we directly call `redirectHYPRLoginPage` instead of initiating this method?

Reply:
This can be done.

Reply:
Fixed with [c32d436](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/c32d436550c773beca2b98f4586420ca1f8bc3c2)